clippy_lints/src/misc.rs

```diff
@@ -519,7 +519,8 @@ fn is_signum(cx: &LateContext<'_, '_>, expr: &Expr) -> bool {
 }
 
 fn is_float(cx: &LateContext<'_, '_>, expr: &Expr) -> bool {
-    matches!(walk_ptrs_ty(cx.tables.expr_ty(expr)).sty, ty::Float(_))
+    let value = walk_ptrs_ty(cx.tables.expr_ty(expr)).sty;
+    matches!(value, ty::Float(_)) || matches!(value, ty::Array(ty::Float, _))
```

Comment:
It should even be possible to write this in a single `matches!` macro:
```suggestion
    matches!(
        value,
        ty::Float(_) | ty::Array(ty::Float(_), _),
    )
```